src/Pact/Types/Typecheck.hs

```diff
@@ -339,7 +339,8 @@ data AST n =
   _aEntity :: Maybe (AST n),
   _aExec :: AST n,
   _aRollback :: Maybe (AST n),
-  _aYieldResume :: Maybe (YieldResume n)
+  _aYieldResume :: Maybe (YieldResume n),
+  _aModel :: ![Exp Info]
```

Comment:
I added this model field to `Step`, which doesn't really follow the existing pattern. However, this seems by far the easiest way to get model information to analysis.

Reply:
It feels a little weird that `Exp` is making it so far down into the "pipeline" but that's how we are representing models right now. This seems fine.

Reply:
Completely agreed